Code/Tools/Libs/ToolsFoundation/NodeObject/Implementation/DocumentNodeManager.cpp

```diff
@@ -268,126 +298,259 @@ void ezDocumentNodeManager::MoveNode(const ezDocumentObject* pObject, const ezVe
+      DocumentNodeManager_NodeMetaData nodeMetaData;
+      rttiConverter.ApplyPropertiesToObject(pAbstractObject, pNodeMetaDataType, &nodeMetaData);
+
+      if (CanMoveNode(pObject, nodeMetaData.m_Pos).m_Result.Succeeded())
       {
-        data.m_NodePos.SetZero();
-        data.m_Connections.Clear();
-        rttiConverter.ApplyPropertiesToObject(pNode, pType, &data);
+        if (bUndoable)
+        {
+          ezMoveNodeCommand move;
+          move.m_Object = pObject->GetGuid();
+          move.m_NewPos = nodeMetaData.m_Pos;
+          history->AddCommand(move);
+        }
+        else
+        {
+          MoveNode(pObject, nodeMetaData.m_Pos);
+        }
+      }
 
-        if (CanMoveNode(pObject, data.m_NodePos).m_Result.Succeeded())
+      // Backwards compatibility to old file format
+      if (auto pOldConnections = pAbstractObject->FindProperty("Node::Connections"))
+      {
+        EZ_ASSERT_DEV(bUndoable == false, "Undo not supported for old file format");
+        RestoreOldMetaDataAfterLoading(graph, *pOldConnections, pObject);
+      }
+    }
+    else if (IsConnection(pObject))
+    {
+      DocumentNodeManager_ConnectionMetaData connectionMetaData;
+      rttiConverter.ApplyPropertiesToObject(pAbstractObject, pConnectionMetaDataType, &connectionMetaData);
+
+      ezDocumentObject* pSource = GetObject(connectionMetaData.m_Source);
+      ezDocumentObject* pTarget = GetObject(connectionMetaData.m_Target);
+      if (pSource == nullptr || pTarget == nullptr)
+        continue;
+
+      const ezPin* pSourcePin = GetOutputPinByName(pSource, connectionMetaData.m_SourcePin);
+      const ezPin* pTargetPin = GetInputPinByName(pTarget, connectionMetaData.m_TargetPin);
+      if (pSourcePin == nullptr || pTargetPin == nullptr)
+        continue;
+
+      ezDocumentNodeManager::CanConnectResult res;
+      if (CanConnect(*pSourcePin, *pTargetPin, res).m_Result.Succeeded())
+      {
+        if (bUndoable)
+        {
+          ezConnectNodePinsCommand cmd;
+          cmd.m_ConnectionObject = pObject->GetGuid();
+          cmd.m_ObjectSource = connectionMetaData.m_Source;
+          cmd.m_ObjectTarget = connectionMetaData.m_Target;
+          cmd.m_sSourcePin = connectionMetaData.m_SourcePin;
+          cmd.m_sTargetPin = connectionMetaData.m_TargetPin;
+          history->AddCommand(cmd);
+        }
+        else
         {
-          if (bUndoable)
-          {
-            ezMoveNodeCommand move;
-            move.m_Object = pObject->GetGuid();
-            move.m_NewPos = data.m_NodePos;
-            history->AddCommand(move);
-          }
-          else
-          {
-            MoveNode(pObject, data.m_NodePos);
-          }
+          Connect(pObject, *pSourcePin, *pTargetPin);
         }
+      }
+    }
+  }
+}
 
-        for (const auto& con : data.m_Connections)
+void ezDocumentNodeManager::GetMetaDataHash(const ezDocumentObject* pObject, ezUInt64& inout_uiHash) const
+{
+  if (IsNode(pObject))
+  {
```

Comment:
Why is this not hashing the node positon?

Reply:
Because the previous implementation didn't hash it either. I just copied the logic from the previous implementations adapted it to the new data structure and moved it to the manager. Do you think it should hash the node position as well?

Reply:
Hash is only used for transform and for that node position is irrelevant. Added a comment in the code.